### ISSUE TITLE
Automatically migrate app project to use the new Android embedding

### ISSUE THREAD

**Issue by blasten** [flutter/flutter#41757]
If an app is using the old embedding and a plugin is using the new embedding. Then, we could try to migrate the app to the new embedding automatically.

**Comment by xster** [comment]
note for future: consider using something like https://github.com/amirh/dart_lsc

**Comment by blasten** [comment]
I chatted with @amirh. He said that  dart_lsc's  scope is about sending PRs, filing issues, and so forth instead of actually performing the migration.  https://flutter.dev/go/flutter-migrate is still the way to streamline migrations affecting project files.

**Comment by xster** [comment]
@amirh didn't you just tell me something different? :D

e.g. each migration recipe (that the breaking change author should maintain (either in the flutter repo or somewhere else)) should implement https://github.com/amirh/dart_lsc/blob/master/packages/migrate_to_1/lib/migrate_base.dart#L39. 

dart_lsc does do a lot more than perform the actual changes but we want to cut that part out so we can execute just that part against the user's own code in flutter migrate.

**Comment by amirh** [comment]
I agree with what both of you are saying 😄 

I guess on a higher level an automated migration is composed of:
1. A delivery vehicle.
1. A migration tool / "recipe".


[dart_lsc](https://github.com/amirh/dart_lsc/tree/master/packages/dart_lsc) is a delivery vehicle, you provide it with a "recipe" (through the `--update_script` argument) and it takes care of applying that change across the ecosystem.

For the current migration I have implemented one migration tool for `dart_lsc` to use. This is the [migrate_to_1](https://github.com/amirh/dart_lsc/tree/master/packages/migrate_to_1) package, in it there's some logic that deals with the `dart_lsc` protocol which can be re-used for future migration tools, and as you mentioned the unique "recipe" part is the [Migration](https://github.com/amirh/dart_lsc/blob/master/packages/migrate_to_1/lib/migrate_base.dart#L39) interface.

The [flutter migrate proposal](https://flutter.dev/go/flutter-migrate) talks about a specific recipe "update project template files" and also about a delivery vehicle for that specific migration (showing warnings in `flutter doctor` to prompt the user to manually run the migration).

We should make sure we have a coherent story around these 2 (e.g that the flutter migrate recipe part is compatible with `dart_lsc`, and potentially that the delivery vehicle for `flutter migrate` can deliver `dart_lsc` migrations like `migrate_to_`).

I'll start a document about this topic.

**Comment by xster** [comment]
Right, whatever we create for 1 should be done in a modular fashion such that the subcomponents are reusable for 2.
